app.py

```diff
@@ -98,7 +98,8 @@ def submit(namespace, doctype, docversion=None, **kwargs):
     try:
         docversion = docversion or SCHEMA_VERSIONS[namespace][doctype]
         key = "{}.{}".format(doctype, docversion)
-        NAMESPACE_SCHEMAS[namespace][key](request.data)
+        json_data = request.get_json(force=True)
```

Comment:
Why parse, then re-serialize the request data here?

Reply:
I think when the header isn't set, then the request shows up as form data. This call is consistent whether the content-type header is set or not.

The rapidjson validator accepts strings, so it has to be re-serialized.

Reply:
Ok, makes sense - can you add a comment to that effect in the code?